float64[] color
float64[] position
float64 q
